contrib/jmx-metrics/src/main/java/io/opentelemetry/contrib/jmxmetrics/JmxMetrics.java

```
@@ -0,0 +1,132 @@
+            new Runnable() {
+              @Override
+              public void run() {
+                try {
+                  runner.run();
+                } catch (Throwable e) {
+                  logger.log(Level.SEVERE, "Error gathering JMX metrics", e);
+                }
+              }
+            }),
+        0,
+        config.intervalMilliseconds);
+    logger.info("Started GroovyRunner.");
+  }
+
+  private static TimerTask wrapTimerTask(final Runnable r) {
+    return new TimerTask() {
+      @Override
+      public void run() {
+        r.run();
+      }
+    };
+  }
+
+  private void shutdown() {
+    logger.info("Shutting down JmxMetrics Groovy runner and exporting final metrics.");
+    timer.cancel();
+    runner.flush();
+  }
+
+  private static JmxConfig getConfigFromArgs(final String[] args) {
+    if (args.length != 0 && (args.length != 2 || !args[0].equalsIgnoreCase("-config"))) {
+      System.out.println(
+          "Usage: java io.opentelemetry.contrib.jmxmetrics.JmxMetrics "
+              + "-config <path_to_config.properties>");
+      System.exit(1);
+    }
+
+    Properties props = new Properties();
+    if (args.length == 2) {
+      try (InputStream is = new FileInputStream(args[1])) {
+        props.load(is);
+      } catch (IOException e) {
+        System.out.println(
+            "Failed to read config properties file at '" + args[1] + "': " + e.getMessage());
+        System.exit(1);
+      }
+    }
+
+    return new JmxConfig(props);
+  }
+
+  /**
+   * Main method to create and run a {@link JmxMetrics} instance.
+   *
+   * @param args - must be of the form "-config jmx_config_path"
+   */
+  public static void main(final String[] args) {
+    JmxConfig config = getConfigFromArgs(args);
+    config.validate();
+
+    final JmxMetrics jmxMetrics = new JmxMetrics();
+    try {
+      jmxMetrics.start(config);
+    } catch (ConfigureError e) {
+      logger.severe(e.getMessage());
+      System.exit(1);
```

Think it's fine to throw the exception instead of logging / exiting